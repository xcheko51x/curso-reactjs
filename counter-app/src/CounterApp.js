import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value = 10}) => {

    const [counter, setCounter] = useState(value); // Hook desestructuracion [valor, funcion _setValor]


    const handleAdd = () => {
        setCounter(counter + 1);
    }

    const handleReset = () => {
        setCounter(value);
    }

    const handleSubtraction = () => {
        setCounter(counter - 1);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>reset</button>
            <button onClick={handleSubtraction}>-1</button>
        </>
    );
}


CounterApp.propTypes = {
    value: PropTypes.number
};

export default CounterApp;