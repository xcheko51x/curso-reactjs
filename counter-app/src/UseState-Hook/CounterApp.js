import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(0); // Hook desestructuracion [valor, funcion _setValor]


    const handleAdd = () => {
        setCounter(counter + 1);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <p>{counter}</p>
            <button onClick={handleAdd}>+1</button>
        </>
    );
}


CounterApp.propTypes = {
    value: PropTypes.number
};

export default CounterApp;