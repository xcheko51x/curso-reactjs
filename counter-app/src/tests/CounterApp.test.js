import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas en el <CounterApp />', () => {

    let wrapper = shallow(<CounterApp />);

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test('Debe mostrar <CounterApp /> correctamente (usando match con snapshot y el valor por defecto)', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar el valor por defecto de 100, usando el wrapper.find, tomando el elemento html donde se muestra el valor del contador', () => {
        
        const wrapper = shallow(<CounterApp value={100} />);

        const counterText = wrapper.find('h2').text().trim();
        
        expect(counterText).toBe('100');

    });

    test('Debe incrementar el contador +1', () => {

        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('11');
     });

     test ('Debe resetear el contador', () => {
        const wrapper = shallow(<CounterApp value={105} />);
        
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('105');
     });

     test('Debe decrementar el contador -1', () => {

        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('9');
     });

});