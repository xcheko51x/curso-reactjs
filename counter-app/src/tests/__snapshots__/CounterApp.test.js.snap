// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> Debe mostrar <CounterApp /> correctamente (usando match con snapshot y el valor por defecto) 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
